File: bhuvigyan-backend/app/schemas/fraud.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime
from uuid import UUID

class FraudFactor(BaseModel):
    name: str
    weight: float
    direction: str
    description: str

class FraudScoreOut(BaseModel):
    model_config = {"protected_namespaces": (), "from_attributes": True}

    id: UUID
    claim_id: UUID
    score: float
    confidence: float
    risk_level: str
    model_version: str
    computed_at: datetime

class FraudExplanationOut(BaseModel):
    model_config = {"from_attributes": True}

    id: UUID
    fraud_score_id: UUID
    top_factors: List[FraudFactor]
    shap_values: Optional[Dict[str, Any]] = None
    human_readable_text: Optional[str] = None

class FraudScoreCreate(BaseModel):
    model_config = {"protected_namespaces": ()}

    claim_id: UUID
    score: float
    confidence: float
    risk_level: str
    model_version: str

class ScoreOverrideRequest(BaseModel):
    override_score: float
    override_reason: str = Field(..., min_length=20)

File: Frontend/bhuvigyan-ui/src/api/fraud.ts
import api from './axios';

export interface FraudScore {
  score: number;
  confidence: number;
  riskLevel: string;
  modelVersion: string;
  explanation?: {
    topFactors: Array<{ name: string; weight: number; direction: string; description: string }>;
    humanReadableText: string;
  };
}

export interface ScoreOverrideRequest {
  overrideScore: number;
  overrideReason: string;
}

export const getFraudScore = async (claimId: string): Promise<FraudScore> => {
  const res = await api.get(`/fraud-scores/${claimId}`);
  return res.data.data;
};

export const overrideFraudScore = async (claimId: string, data: ScoreOverrideRequest) => {
  const res = await api.post(`/fraud-scores/${claimId}/override`, data);
  return res.data;
};

export const triggerScoring = async (claimId: string) => {
  const res = await api.post(`/admin/claims/${claimId}/score`);
  return res.data;
};
